crates/accelerate/src/euler_one_qubit_decomposer.rs

```diff
@@ -688,15 +683,13 @@ fn mod_2pi(angle: f64, atol: f64) -> f64 {
 }
 
 fn params_zyz_inner(mat: ArrayView2<Complex64>) -> [f64; 4] {
-    let coeff: Complex64 = 1. / det_one_qubit(mat).sqrt();
-    let phase = -complex_phase(coeff);
-    let tmp_1_0 = (coeff * mat[[1, 0]]).abs();
-    let tmp_0_0 = (coeff * mat[[0, 0]]).abs();
-    let theta = 2. * tmp_1_0.atan2(tmp_0_0);
-    let phiplambda2 = complex_phase(coeff * mat[[1, 1]]);
-    let phimlambda2 = complex_phase(coeff * mat[[1, 0]]);
-    let phi = phiplambda2 + phimlambda2;
-    let lam = phiplambda2 - phimlambda2;
+    let det = det_one_qubit(mat);
+    let phase = 0.5 * det.arg();
+    let theta = 2. * mat[[1, 0]].abs().atan2(mat[[0, 0]].abs());
+    let ang1 = mat[[1, 1]].arg();
+    let ang2 = mat[[1, 0]].arg();
+    let phi = ang1 + ang2 - det.arg();
+    let lam = ang1 - ang2;
```

Comment:
Can remove one call to arg() since `det` itself is not needed?
```suggestion
    let det_arg = det_one_qubit(mat).arg();
    let phase = 0.5 * det_arg;
    let theta = 2. * mat[[1, 0]].abs().atan2(mat[[0, 0]].abs());
    let ang1 = mat[[1, 1]].arg();
    let ang2 = mat[[1, 0]].arg();
    let phi = ang1 + ang2 - det_arg;
    let lam = ang1 - ang2;
```

Reply:
In theory LLVM should detect that `Complex64::arg` is a pure function and optimise out all but the first call, but it doesn't hurt to be more explicit.

Reply:
oh, that's cool, thanks for educating me. Probably not worth the edit then.

Reply:
Oh, I still think it's a good idea to put in - no particular reason to rely on LLVM noticing it and doing it when it's easy for us to just be explicit.